### PR TITLE
Upgrade django to v1.8.

### PR DESCRIPTION
**Pull request by muzaffaryousaf**
TNL-2822

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # Django/Framework Packages
-django==1.4.12
+django>=1.8,<1.9
```
Comment (nedbat) [inline requirements.txt]: We should pin to a particular 1.8 version.

Comment (nedbat) [inline requirements.txt]: Oops, my mistake, this is a library, not an IDA, so we should not pin a version.

File: milestones/migrations/0001_initial.py [hunk 2]
```diff
@@ -1,168 +1,147 @@
...
+from django.db import models, migrations
+import django.utils.timezone
+import model_utils.fields
+from milestones.data import fetch_milestone_relationship_types
+
+# See this link for forwards_func and reverse_func
+# https://docs.djangoproject.com/en/1.8/ref/migration-operations/#runpython
+
+
+def forwards_func(apps, schema_editor):
```
Comment (muzaffaryousaf) [inline milestones/migrations/0001_initial.py]: *Reviewers:* This is added by me manually for seeding data.

File: milestones/migrations/0001_initial.py [hunk 3]
```diff
@@ -1,168 +1,147 @@
...
+        try:
+            milestone_relationship_type_model.objects.using(db_alias).get(name=name)
+        except milestone_relationship_type_model.DoesNotExist:
+            milestone_relationship_type_model.objects.using(db_alias).create(
+                name=name,
+                description='Autogenerated milestone relationship type "{}"'.format(name),
+            )
+
+
+def reverse_func(apps, schema_editor):
```
Comment (muzaffaryousaf) [inline milestones/migrations/0001_initial.py]: Same as forward.

File: milestones/migrations/0001_initial.py [hunk 4]
```diff
@@ -1,168 +1,147 @@
...
+        ),
+        migrations.AlterUniqueTogether(
+            name='coursemilestone',
+            unique_together=set([('course_id', 'milestone')]),
+        ),
+        migrations.AlterUniqueTogether(
+            name='coursecontentmilestone',
+            unique_together=set([('course_id', 'content_id', 'milestone')]),
+        ),
+        migrations.RunPython(forwards_func, reverse_func, False),
```
Comment (muzaffaryousaf) [inline milestones/migrations/0001_initial.py]: Check the docs [here](https://docs.djangoproject.com/en/1.8/ref/migration-operations/#runpython).

Comment (muzaffaryousaf) [inline milestones/migrations/0001_initial.py]: `forwards_func, reverse_func, False`  False is for atomic, please check the link above.

File: milestones/tests/utils.py [hunk 5]
```diff
@@ -20,7 +20,7 @@ def setUp(self):
         self.test_course_key = CourseKey.from_string('the/course/key')
         self.test_prerequisite_course_key = CourseKey.from_string('the/prerequisite/key')
         self.test_content_key = UsageKey.from_string('i4x://the/content/key/12345678')
-        self.test_user = User.objects.create(
+        self.test_user = User.objects.create_user(
```
Comment (muzaffaryousaf) [inline milestones/tests/utils.py]: This is to avoid the no-member pylint warning.

**Comment by muzaffaryousaf** [comment]
@symbolist @wedaly @macdiesel @muhammad-ammar  Can you guys give it a look. Thanks.

**Comment by muzaffaryousaf** [comment]
@mattdrayer Can you please give it a look as an author of this repo, specific to migration files. ? 
Thanks.

**Comment by muzaffaryousaf** [comment]
@nedbat @symbolist @mattdrayer @wedaly @macdiesel  FYI, this is ready to review.
Thanks.

File: setup.py [hunk 6]
```diff
@@ -23,9 +23,8 @@
     dependency_links=[
```
Comment (macdiesel) [inline setup.py]: We should bump the version number of this library.

**Comment by symbolist** [comment]
:+1:

**Comment by mattdrayer** [comment]
Sorry I'm just now jumping into this PR -- it seems to me there are still some open issues noted in the PR thread before we can consider this particular changeset ready to merge -- is that correct?

**Comment by muzaffaryousaf** [comment]
@mattdrayer Yes they are not addressed yet and this is not ready to merge yet, will address those when we are done with edx-platform upgrade.

**Pull request merged**